docs/how-to-guides/use-completion.rst

```diff
@@ -0,0 +1,58 @@
+.. _use_completion:
+
+Use the completion module
+=========================
+
+Craft-cli provides a completion module for auto-generating bash completion scripts for
+applications using its :py:class:`Dispatcher`.
+
+In order to invoke it, the application needs to have a public function that returns
+some basic information about itself. All applications must provide their
+:py:class:`Dispatcher` and a configuration to initialise commands with. By default,
+craft-cli commands don't need to be initialised with anything, so this would be
+``None`` in the basic case.
```

Comment:
It took me a while to realise the configuration we're referring to is the `None` in the function's return.

I think we could make this segment clearer by:
- Clarifying what we're going to do with this app getter.
- Discussing more plainly what the developer will do with the configuration.
- In the absence of a reference, describe the relationship between the module and the dispatcher.